src/nomad_simulation_parsers/parsers/h5md/parser.py

```diff
@@ -0,0 +1,358 @@
+from importlib import reload
+from typing import Any
+
+import pint
+from nomad.datamodel import EntryArchive
+from nomad.parsing.file_parser import ArchiveWriter
+from nomad.parsing.file_parser.mapping_parser import HDF5Parser, MetainfoParser, Path
+from nomad.units import ureg
+from nomad.utils import get_logger
+from simulationworkflowschema.molecular_dynamics import MolecularDynamics
+from structlog.stdlib import BoundLogger
+
+from nomad_simulation_parsers.parsers.utils.general import remove_mapping_annotations
+from nomad_simulation_parsers.parsers.utils.mdparserutils import MDParser
+from nomad_simulation_parsers.schema_packages import h5md
+from nomad_simulation_parsers.schema_packages.h5md import Simulation
+
+LOGGER = get_logger(__name__)
+# Make base class non-abstract by providing the property globally
+if not hasattr(MetainfoParser, 'logger'):
+    MetainfoParser.logger = property(lambda self: LOGGER)
+
+
+class H5MDMetainfoParser(MetainfoParser):
+    # TODO: temporary fix until structlog propagation lands everywhere
+    @property
+    def logger(self):
+        return LOGGER
+
+
+class H5MDH5Parser(HDF5Parser):
+    trajectory_steps: list[int] = []
+    output_steps: list[int] = []
+    observables: dict[str, Any] = {}
+
+    # TODO temporary fix for structlog unable to propagate logger
+    @property
+    def logger(self):
+        return LOGGER
+
+    def get_value(self, name: str, dct: dict[str, Any]) -> Any:
+        value = dct.get(name, {})
+        if not isinstance(value, dict):
+            return value
+        value = value.get(self.value_key)
+        if value is None:
+            return
+        unit = dct.get(name, {}).get(f'{self.attribute_prefix}unit')
+        if unit:
+            value = value * ureg(unit)
+        factor = dct.get(name, {}).get(f'{self.attribute_prefix}unit_factor')
+        if factor:
+            value = value * factor
+        return value
+
+    def get_source(self, parent: dict[str, Any], path: str) -> Any:
+        if path is None:
+            return {}
+
+        path_segments = path.split('.', 1)
+        source = parent.get(path_segments[0], {})
+
+        if len(path_segments) == 1:
+            return source
+        return self.get_source(source, path_segments[1])
+
+    def map_value(self, source: dict[str, Any], **kwargs) -> Any:
```

Comment:
a little nitpicking but i think you can put key and enum_spec directly as args

Reply:
you can actually make enum_spec as a function so you do not have the if in l74-77 also more general

Reply:
instead of enum_spec  as string, have something like enum_spec: lambda x: x.upper()

Reply:
ok, I just pass them now as args :sweat_smile: